File: 2020/day2.py
import re

file = open('2020/inputs/day2.txt', "r")
line = file.readline()

validPasswords = 0
while line:
    line = line.replace(':', '')
    data = line.split()
    indexes = data[0].split('-')
    if(data[2].count(data[1]) >= int(indexes[0]) and data[2].count(data[1]) <= int(indexes[1])): validPasswords+=1
    line = file.readline() # Next line
file.close()
print("=============",validPasswords,"=============")

# ====================== day 2 : part 2 ====================== #

import re

file = open('2020/inputs/day2.txt', "r")
line = file.readline()

validPasswords = 0
while line:
    line = line.replace(':', '')
    data = line.split()
    indexes = data[0].split('-')
    if(data[2][int(indexes[0])-1] == data[1] and data[2][int(indexes[1])-1] != data[1]): validPasswords+=1
    elif(data[2][int(indexes[0])-1] != data[1] and data[2][int(indexes[1])-1] == data[1]): validPasswords+=1
    line = file.readline() # Next line
file.close()
print("=============",validPasswords,"=============")

File: 2021/day2.php
<?php
// $file = fopen("inputs/day2.txt", "r");
// if ($file) {
//     $horizontal = $depth = 0;
//     while (($line = fgets($file)) !== false) {
//         $data = explode(" ", $line);
//         switch($data[0]){
//             case 'up':
//                 $depth-=$data[1];
//                 break;
//             case 'down':
//                 $depth+=$data[1];
//                 break;
//             default :
//                 $horizontal+=$data[1];
//         }
//     }
//     echo($horizontal*$depth);
//     fclose($file);
// }

// ====================== day 2 : part 2 ====================== \\

$file = fopen("inputs/day2.txt", "r");
if ($file) {
    $depth = $horizontal = $aim = 0;
    while (($line = fgets($file)) !== false) {
        $data = explode(" ", $line);
        switch($data[0]){
            case 'up':
                $aim-=$data[1];
                break;
            case 'down':
                $aim+=$data[1];
                break;
            default :
                $horizontal+=$data[1];
                $depth+=$aim*$data[1];
        }
    }
    echo($horizontal*$depth);
    fclose($file);
}
